tower-http/src/cors/allow_origin.rs

```diff
@@ -129,14 +129,15 @@ impl From<Vec<HeaderValue>> for AllowOrigin {
 
 #[derive(Clone)]
 enum OriginInner {
-    Const(Option<HeaderValue>),
+    Const(HeaderValue),
+    List(Vec<HeaderValue>),
```

Comment:
Maybe

```suggestion
    List(Arc<[HeaderValue]>),
```

Reply:
Why? Do you expect the layer to be cloned commonly?

Reply:
The layer no but the service contains a layer, so yes.

Reply:
And the service will be cloned a lot? Why? Also maybe we should have a single `Arc` around the current fields of the `Layer` then? This seems unrelated in any case.

Reply:
Yes services are often cloned to get an owned value that can be moved into the response future. It doesn't have to happen in this PR but is something to keep in mind.